airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/error_handlers/error_handler.py

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Optional, Union
+
+import requests
+
+
+class ResponseAction(Enum):
+    """
+    Response statuses for non retriable responses
+    """
+
+    SUCCESS = "SUCCESS"  # "Request was successful"
+    FAIL = "FAIL"  # "Request failed unexpectedly"
+    IGNORE = "IGNORE"  # "Request failed but can be ignored"
+    RETRY = "RETRY"  # Request failed and should be retried
+
+
+class ResponseStatus:
+    """
+    ResponseAction amended with backoff time if a action is RETRY
+    """
+
+    def __init__(self, response_action: Union[ResponseAction, str], retry_in: Optional[float] = None):
+        """
+        :param response_action: response action to execute
+        :param retry_in: backoff time (if action is RETRY)
+        """
+        if isinstance(response_action, str):
+            response_action = ResponseAction[response_action]
+        if retry_in:
+            assert response_action == ResponseAction.RETRY
+        self._retry_in = retry_in
+        self._action = response_action
+
+    @property
+    def action(self):
+        return self._action
+
+    @property
+    def retry_in(self) -> Optional[float]:
+        return self._retry_in
+
+    @classmethod
+    def success(cls):
```

Comment:
I used classmethods to prevent us from mutating the objects, but I found the `Final` [type annotation](https://docs.python.org/3/library/typing.html#typing.Final) so they're just global `Final` values now https://github.com/airbytehq/airbyte/pull/14330/files#diff-c3f497e9893f415a917c58874ce9af4ac0ceb5c71b570f65298093354464674bR48-R50